### PR TITLE
Utilize DownloadableDataset

### LINKED ISSUE #22: Utilize cxflow DownloadableDataset

### PR DESCRIPTION
**Pull request by SvecDavid**
closes #22

### CODE REVIEW

File: mnist_convnet/mnist_dataset.py [hunk 1]
```diff
@@ -15,15 +16,24 @@
...
 
-    def _configure_dataset(self, data_root=path.join('datasets', '.mnist-data'), batch_size:int=100, **kwargs) -> None:
+    def _configure_dataset(self, data_root=path.join('mnist_convnet', '.mnist-data'), batch_size:int=100, **kwargs) -> None:
         self._batch_size = batch_size
         self._data_root = data_root
+        self._download_urls = [path.join(DOWNLOAD_ROOT, filename) for filename in FILENAMES.values()]
         self._data = {}
         self._data_loaded = False
 
+    @property
```
Comment (blazekadam) [inline mnist_convnet/mnist_dataset.py]: Lines 29 to 36 can be remove. The properties are inherited from the ancestor.

File: mnist_convnet/mnist_dataset.py [hunk 2]
```diff
@@ -15,15 +16,24 @@
...
 
 
-class MNISTDataset(cx.BaseDataset):
+class MNISTDataset(cx.DownloadableDataset):
     """ MNIST dataset for hand-written digits recognition."""
 
-    def _configure_dataset(self, data_root=path.join('datasets', '.mnist-data'), batch_size:int=100, **kwargs) -> None:
+    def _configure_dataset(self, data_root=path.join('mnist_convnet', '.mnist-data'), batch_size:int=100, **kwargs) -> None:
         self._batch_size = batch_size
         self._data_root = data_root
```
Comment (blazekadam) [inline mnist_convnet/mnist_dataset.py]: lines 24 and 25 can be substituted with `super().__init__(...)` call but this is good as well

File: mnist_convnet/mnist_dataset.py (hunk 1 above)
Comment (blazekadam) [reply mnist_convnet/mnist_dataset.py]: (Do not forget to remove the typing import as well).

Comment (SvecDavid) [reply mnist_convnet/mnist_dataset.py]: These methods are defined as abstractmethod in DownloadableDataset, so I need to override them.

Comment (blazekadam) [reply mnist_convnet/mnist_dataset.py]: I see, that does not make much sense actually. Why they are abstract if they have an implementation? Can you fix it in *cxflow* and remove it here?

Comment (SvecDavid) [reply mnist_convnet/mnist_dataset.py]: Yes, I will do it.

**Pull request merged**